core/src/main/java/bisq/core/dao/node/explorer/ExportJsonFilesService.java

```diff
@@ -113,57 +102,86 @@ public void start() {
+        DaoState daoState = daoStateService.getClone();
+        List<Block> blocks;
+        if (optionalHeight.isPresent()) {
+            int height = Integer.parseInt(optionalHeight.get());
+            blocks = daoState.getBlocks().stream()
+                    .filter(block -> block.getHeight() > height)
                     .collect(Collectors.toList());
-            JsonBlocks jsonBlocks = new JsonBlocks(daoState.getChainHeight(), jsonBlockList);
-
-            ListenableFuture<Void> future = executor.submit(() -> {
-                bsqStateFileManager.writeToDisc(Utilities.objectToJson(jsonBlocks), "blocks");
-                allJsonTxOutputs.forEach(jsonTxOutput -> txOutputFileManager.writeToDisc(Utilities.objectToJson(jsonTxOutput), jsonTxOutput.getId()));
-                jsonTxs.forEach(jsonTx -> txFileManager.writeToDisc(Utilities.objectToJson(jsonTx), jsonTx.getId()));
-                return null;
-            });
-
-            Futures.addCallback(future, Utilities.failureCallback(throwable -> {
-                log.error(throwable.toString());
-                throwable.printStackTrace();
-            }), MoreExecutors.directExecutor());
+        } else {
+            blocks = daoState.getBlocks();
+        }
+
+        if (!blocks.isEmpty()) {
+            process(blocks, daoState);
         }
     }
 
+    private void process(List<Block> blocks, DaoState daoState) {
+        int chainHeight = daoState.getChainHeight();
+        List<JsonBlock> jsonBlockList = blocks.stream()
+                .map(this::getJsonBlock)
+                .collect(Collectors.toList());
+
+        List<JsonTxOutput> jsonTxOutputs = new ArrayList<>();
+        List<JsonTx> jsonTxs = blocks.stream()
+                .flatMap(block -> block.getTxs().stream())
+                .map(tx -> {
+                    JsonTx jsonTx = getJsonTx(tx);
+                    jsonTxOutputs.addAll(jsonTx.getOutputs());
+                    return jsonTx;
+                })
+                .collect(Collectors.toList());
+
+        ListenableFuture<Void> future = executor.submit(() -> {
+            log.info("Write chainHeight={}, jsonBlockList={}, jsonTxs={}, jsonTxOutputs={}",
+                    chainHeight, jsonBlockList.size(), jsonTxs.size(), jsonTxOutputs.size());
+            chainHeightFileManager.write(String.valueOf(chainHeight), "chainHeight");
+            jsonBlockList.forEach(jsonBlock -> blocksFileManager.write(Utilities.objectToJson(jsonBlock), jsonBlock.getHeight() + "_" + jsonBlock.getHash()));
+            jsonTxs.forEach(jsonTx -> txFileManager.write(Utilities.objectToJson(jsonTx), jsonTx.getId()));
+            jsonTxOutputs.forEach(jsonTxOutput -> txOutputFileManager.write(Utilities.objectToJson(jsonTxOutput), jsonTxOutput.getId()));
+            return null;
+        });
+
+        Futures.addCallback(future, Utilities.failureCallback(throwable -> {
+            log.error(throwable.toString());
+            throwable.printStackTrace();
```

Comment:
Isn't this t.printStackTrace() redundant?  You just logged the error, with stack trace.